js/src/components/free-listings/configure-product-listings/shipping-time/shipping-time-setup/add-time-button/add-time-modal/index.js

```diff
@@ -25,38 +26,21 @@ import SupportedCountrySelect from '.~/components/supported-country-select';
 const AddTimeModal = ( { countries, onRequestClose, onSubmit } ) => {
 	const [ dropdownVisible, setDropdownVisible ] = useState( false );
 
-	const handleValidate = ( values ) => {
-		const errors = {};
-
-		if ( values.countries.length === 0 ) {
-			errors.countries = __(
-				'Please specify at least one country.',
-				'google-listings-and-ads'
-			);
-		}
-
-		if ( values.time < 0 ) {
-			errors.time = __(
-				'The estimated shipping time cannot be less than 0.',
-				'google-listings-and-ads'
-			);
-		}
-
-		return errors;
-	};
-
 	const handleSubmitCallback = ( values ) => {
-		onSubmit( values );
+		onSubmit( {
+			countries: values.countryCodes,
```

Comment:
> Wondering if these two names would gradually converge towards one of them?

Yeah, I think the ideal situation is that everything converges towards one of them, across shipping rates and shipping times, in both back-end PHP and front-end JS code.

For some background context, the `country_code` parameter in the shipping rate API was renamed to `country` in PR https://github.com/woocommerce/google-listings-and-ads/pull/1202. That's why on the shipping rate JS side, we have `country` / `countries`, to match with the API.

Meanwhile, in shipping time API, we still have `country_code` parameter, that's why we have `countryCode` / `countryCodes` in shipping time JS side.

My preference is to make the naming in JS closely match with the naming in API side.

If we want, we can standardize them to `countries` /  on the JS side, and map `countries` to `country_codes` when we are making API calls. IMO, this is less than ideal, but doable. Let me know your thoughts.

> This PR adjusted some of them but there are still some inconsistent naming within individual components. The current implementation makes the switching between `countries` and `countryCodes` within and between components more complicated, resulting in lower readability.

Yeah, agree. It was quite a deep rabbit hole when I first looked into this issue, I tried to make them nice and consistent but there are too many changes, IMO way too much to fix this issue #1435. In the end, I had to "restart" and make the solution focusing on fixing issue #1435, without looking into other engineering / tech debt concern.  

> If at least these shipping modals interface, internal and its dependencies are adjusted to be consistent in this PR first, it might save time for future code reviews.

I looked into that, and it causes a lot of code changes. 

So the idea that I have here is that I would like this PR to focus on fixing issue #1435, so reviewers could (relatively easily) verify that the fix here does fix the issue, without introducing other issues.

Next, we can have a separate PR to address the engineering / tech debt concern, so reviewers could focus on the engineering changes, and no changes to functionalities.

I'm thinking I can address these engineering concerns when I work on refactoring the shipping time section. 📜

Reply:
📜 

> My preference is to make the naming in JS closely match with the naming in API side.

Considering that JS and PHP have different naming in their coding styles, the ideal naming consistency is unlikely to be achieved. Even if this ideal is achievable, I think the strategy of making the naming in JS closely match the API side is a secondary consideration. Having an isolation layer allows for renaming without dripping into the wider code, and allows for flexibility on JS side of the interface design.

> If we want, we can standardize them to `countries` / on the JS side, and map `countries` to `country_codes` when we are making API calls. IMO, this is less than ideal, but doable. Let me know your thoughts.

Yes, this's the way to create an isolation layer.

The isolation layer could be explicitly established between the **wp-data** and the application code, so that if naming consistency or naming mapping is needed, it could be adapted before the fetched data is passed to the **reducer**. Then as you said, when using **actions** to communicate the API, internally map the parameters of the action function to API spec.

---

💅 

> It was quite a deep rabbit hole when I first looked into this issue, I tried to make them nice and consistent but there are too many changes, IMO way too much to fix this issue #1435. In the end, I had to "restart" and make the solution focusing on fixing issue #1435, without looking into other engineering / tech debt concern.

I was thinking it would be good to address some technical debts together within certain scope when fixing the problem or adding new features. Because sometimes if we want to continue modifying the code without dealing with the technical debt, we may therefore be limited by it, unable to use better practices, or even worse, add more technical debt.

After reading through the larger code scope, I understand that it's difficult to make the consistent `country` naming in components within an explicit scope, and that eventually, some components will still have to bear those naming mapping first.

> > If at least these shipping modals interface, internal and its dependencies are adjusted to be consistent in this PR first, it might save time for future code reviews.
> 
> I looked into that, and it causes a lot of code changes.
> 
> The idea that I have here is that I would like this PR to focus on fixing issue #1435, so reviewers could (relatively easily) verify that the fix here does fix the issue, without introducing other issues.
> 
> I'm thinking I can address these engineering concerns when I work on refactoring the shipping time section.

As a code reviewer, I had to check the mapping of these names back and forth in several similar implementations to make sure they are all not missed. And originally I was not sure which of the two variable names would be converged toward, so I was worried that when adjusting the shipping time section, the naming of `countries` or `countryCode` would be changed again where this PR has been changed, so there would be extra review time spent.

If the naming in shipping time would also be changed to `countries`, then maybe the countries -> countryCode mapping direction of this PR could be changed to countryCode -> countries, then it should still save some scope and time for future adjustment.

Reply:
> The isolation layer could be explicitly established between the **wp-data** and the application code [...]

Yup. We have the same thinking here. 🙂 👍 

> I was thinking it would be good to address some technical debts together within certain scope when fixing the problem or adding new features. [...]

Yup, agree. In this PR, I already see myself as doing some form of "address some technical debts together within certain scope", by refactoring the shipping time validation into a function shared by the different shipping time UI pieces, and adding tests for the validation function. 😄 

> As a code reviewer, I had to _check the mapping of these names back and forth_ in several similar implementations to make sure they are all not missed. And originally I was not sure _which of the two variable names would be converged toward_, so I was worried that when adjusting the shipping time section, the naming of `countries` or `countryCode` would be changed again where this PR has been changed, so there would be extra review time spent.
> 
> _If the naming in shipping time would also be changed to `countries`_, then _maybe the countries -> countryCode mapping direction of this PR could be changed to countryCode -> countries_, then it should still save some scope and time for future adjustment.

Hey, thanks a lot for this comment here! 😃 👍  Combined with my earlier comment https://github.com/woocommerce/google-listings-and-ads/pull/1473#discussion_r865028069 about "we can standardize them to `countries` on the JS side, and map `countries` to `country_codes` when we are making API calls", I think I realize what you are saying. When I take another look into `validateShippingRateGroup` and `validateShippingTimeGroup`, one is using `countries` and another is using `countryCodes`, which can be a bit confusing. In the future when we work on refactoring shipping time section, I'm leaning towards making them into `countries`, and it make sense for us to change to using `countries` in both `validateShippingRateGroup` and `validateShippingTimeGroup` in this PR. I have made the change in [`b505b33` (#1473)](https://github.com/woocommerce/google-listings-and-ads/pull/1473/commits/b505b33080e31512fa58e428a6a9c2e0fadf2d0d).